include/gsl/gsl-lite.hpp

```diff
@@ -1742,7 +1742,7 @@ typedef gsl_CONFIG_INDEX_TYPE diff;
 #if gsl_DEVICE_CODE
 # if defined( gsl_CONFIG_DEVICE_UNENFORCED_CONTRACTS_ASSUME )
 #  if gsl_COMPILER_NVCC_VERSION >= 113
-#   define gsl_ASSUME_( x )           ( __builtin_assume( x ) )
+#   define gsl_ASSUME_( x )           ( __builtin_assume( bool(x) ) )
```

Comment:
Function-style casts are equivalent to C casts, i.e. `(bool) (x)`, which are discouraged. Please use the idiomatic conversion-by-double-negation instead: `!!( x )`.